### DISCUSSION TITLE
How to turn off all the access logs of longhorn-manager due to readiness probes ?

### DISCUSSION THREAD

**Discussion by ThaiDangFr** [longhorn/longhorn#9412] [General]
Dear all,

daemonset longhorn-manager is producing a lot of access logs like this :

`10.42.1.137 - - [06/Sep/2024:09:39:03 +0000] "GET /v1/volumes/pvc-b3493b6c-226f-48ec-903d-4a9942428f55 HTTP/1.1" 200 10684 "" "Go-http-client/1.1"
10.42.1.137 - - [06/Sep/2024:09:39:03 +0000] "GET /v1/volumes/pvc-b8baca38-6757-4bdd-bb1a-62f609cb8fcd HTTP/1.1" 200 12068 "" "Go-http-client/1.1"
10.42.1.137 - - [06/Sep/2024:09:39:11 +0000] "GET /v1/volumes/pvc-8bba0d90-0ba3-4121-96d2-f5896355eafd HTTP/1.1" 200 12074 "" "Go-http-client/1.1"`

is there a way to turn that off ?

**Comment by PhanLe1010** [comment]
It is the logs when the CSI plugin or recurring job query the Longhorn API. 

I am not able to find a way to disable it. Any idea @ejweber @ChanYiLin ?

If your cluster is pretty idle it might be the case longhorn-manager mostly contains these logs. But they shouldn't spam the longhorn manager too often. How frequently do they appear?

**Reply by ChanYiLin** [reply]
Operations via UI might also introduce these logs.

It is because we use https://pkg.go.dev/github.com/gorilla/handlers#CombinedLoggingHandler to log alll the access requests of the http server.
Here is the commit.
https://github.com/longhorn/longhorn-manager/pull/235/commits/7832a879c0ad8350eb6c616491e9d0554872ebbf

I am not sure if there is an easy way to disable it as it is a middleware when setting up the http server.

**Reply by ejweber** [reply]
Agreed. Checking the code @ChanYiLin linked, it looks like it would be pretty easy for us to choose NOT to use that middleware as the result of some user-configurable setting, but in current versions of Longhorn, it's not possible to disable it. TBH, I'm not sure about adding another setting, as I think @PhanLe1010 is right that we usually only see a lot of logs like this build up over time in clusters that don't have anything else interesting going on. (I think a a large number of settings starts to become problematic from a usability standpoint.) I'm also curious how frequently you're setting them @ThaiDangFr.

**Reply by PhanLe1010** [reply]
Thanks @ChanYiLin and @ejweber !

**Comment by ThaiDangFr** [comment]
@ejweber I have 130 lines of these logs every 15 minutes. The longhorn-manager container is producing them. I have 13 volumes used by pods, my cluster is composed of 1 kubernetes master node and 2 worker nodes, with longhorn 1.6.2.

Apart from this log issue, it is working great. As I use a centralized log collector, I try to lower the log volume to essential informations like warnings and errors.

And also, when I am using the longhorn UI, I have these warnings showing up, is it something I need to worry about or is it just for  information purpose ?
```bash
2024-09-11 19:01:51.418	
time="2024-09-11T17:01:51Z" level=warning msg="websocket: close 1000 (normal)" func=api.NewRouter.NewStreamHandlerFunc.func162.1 file="stream.go:52" id=2070498177091790989 type=events
2024-09-11 18:59:50.654	
time="2024-09-11T16:59:50Z" level=warning msg="websocket: close 1000 (normal)" func=api.NewRouter.NewStreamHandlerFunc.func135.1 file="stream.go:52" id=2947801848868600739 type=recurringjobs
2024-09-11 18:59:50.649	
time="2024-09-11T16:59:50Z" level=warning msg="websocket: close 1000 (normal)" func=api.NewRouter.NewStreamHandlerFunc.func144.1 file="stream.go:52" id=6985261372226096280 type=engineimages
2024-09-11 18:59:50.647	
time="2024-09-11T16:59:50Z" level=warning msg="websocket: close 1000 (normal)" func=api.NewRouter.NewStreamHandlerFunc.func147.1 file="stream.go:52" id=3875065992866633609 type=backingimages
2024-09-11 18:59:50.643	
time="2024-09-11T16:59:50Z" level=warning msg="websocket: close 1000 (normal)" func=api.NewRouter.NewStreamHandlerFunc.func129.1 file="stream.go:52" id=9083715667411967623 type=settings
2024-09-11 18:59:46.146	
time="2024-09-11T16:59:46Z" level=warning msg="websocket: close 1000 (normal)" func=api.NewRouter.NewStreamHandlerFunc.func162.1 file="stream.go:52" id=8771051584004514736 type=events
```

**Reply by PhanLe1010** [reply]
Can you config your log collector to ignore/not collect the HTTP request logs from longhorn-manager to reduce the noise? While this info may not be very useful in a healthy cluster, it might be useful when troubleshooting a problematic cluster so we might prefer to keep it as it is for now. 

`websocket: close 1000 (normal)` I think this is normal when you switch to a different tap of Longhorn UI, it might try to create new websockets for different resources of that tap and close the sockets of resources of old tab

**Comment by ThaiDangFr** [comment]
So for now there is no setting that I can use to turn that off deliberately ?
I already tried to put in the setting log-level = Warn, I have less logs which is great, but still have these GET /v1/volumes/pvc lines.
